### PR TITLE
eng, migrate to tsp-client; update compiler 0.53

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
code https://github.com/Azure/autorest.java/pull/2552/commits/3b71c48b47b8f965709f2ee7f0519e9e371185ba

tested in devops

Also sync versions from SDK repo.

### CODE REVIEW

File: .github/workflows/fluent-test-gen-all.yml [hunk 1]
```diff
@@ -8,6 +8,14 @@ jobs:
   build:
     timeout-minutes: 180
     runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: write
+      deployments: none
+      packages: none
+      pull-requests: none
+      security-events: none
```
Comment (weidongxu-microsoft) [inline .github/workflows/fluent-test-gen-all.yml:17]: fix the github bot permission
https://docs.opensource.microsoft.com/github/apps/permission-changes/

File: eng/sdk/sync_sdk.py [hunk 2]
```diff
@@ -71,16 +71,7 @@ def update_sdks():
...
-        cmd = [
-            'pwsh',
-            os.path.join(sdk_root, 'eng/common/scripts/TypeSpec-Project-Sync.ps1'),
-            module_path
-        ]
-        subprocess.check_call(cmd, cwd=sdk_root)
-
-        cmd[1] = os.path.join(sdk_root, 'eng/common/scripts/TypeSpec-Project-Generate.ps1')
-        subprocess.check_call(cmd, cwd=sdk_root)
+        subprocess.check_call(['tsp-client', 'update'], cwd=module_path)
```
Comment (haolingdong-msft) [inline eng/sdk/sync_sdk.py:74]: Is there document about what does `tsp-client update` do? to update typespec dependencies to latest version and generate sdk?

Comment (weidongxu-microsoft) [reply eng/sdk/sync_sdk.py:74]: https://github.com/Azure/azure-sdk-tools/tree/main/tools/tsp-client

Comment (weidongxu-microsoft) [reply eng/sdk/sync_sdk.py:74]: BTW, I've updated Quickstart as well https://github.com/Azure/azure-sdk-for-java/wiki/TypeSpec-Java-Quickstart#use-typespec-defined-in-rest-api-specifications

Comment (haolingdong-msft) [reply eng/sdk/sync_sdk.py:74]: ok, I might miss some context. What is the relationship between nightly sdk preview pipeline and this PR's `tsp-client update`?

Comment (weidongxu-microsoft) [reply eng/sdk/sync_sdk.py:74]: Nothing related at all.

tsp-client just replaces script `TypeSpec-Project-*` https://github.com/Azure/azure-sdk-tools/tree/main/eng/common/scripts

Comment (haolingdong-msft) [reply eng/sdk/sync_sdk.py:74]: ok, I see.

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**